src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -105,7 +105,8 @@ internal sealed class ExternalErrorDiagnosticUpdateSource : IDiagnosticUpdateSou
             _diagnosticService = diagnosticService;
             _buildOnlyDiagnosticsService = _workspace.Services.GetRequiredService<IBuildOnlyDiagnosticsService>();
 
-            _notificationService = _workspace.Services.GetRequiredService<IGlobalOperationNotificationService>();
+            // We are in the VS layer, so getting the IGlobalOperationNotificationService must succeed.
+            _notificationService = _workspace.Services.SolutionServices.ExportProvider.GetExports<IGlobalOperationNotificationService>().Single().Value;
```

Comment:
It looks like the constructor is run here:

https://github.com/dotnet/roslyn/blob/f26bc3080b6e26c196ade50eb537c6b72e769c35/src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs#L166

So you could just pass this through the constructor which then avoids having to grab the ExportProvider from SolutionServices, which seems a bit sketchy.

Reply:
will do!